types/jwt.test-d.ts

```diff
@@ -10,19 +10,19 @@ const secretOptions = {
     public: 'publicKey',
     private: 'privateKey'
   },
-  secretFnCallback: (_req, _token, cb) => { cb(null, 'supersecret') },
-  secretFnPromise: (_req, _token) => Promise.resolve('supersecret'),
-  secretFnAsync: async (_req, _token) => 'supersecret',
-  secretFnBufferCallback: (_req, _token, cb) => { cb(null, Buffer.from('some secret', 'base64')) },
-  secretFnBufferPromise: (_req, _token) => Promise.resolve(Buffer.from('some secret', 'base64')),
-  secretFnBufferAsync: async (_req, _token) => Buffer.from('some secret', 'base64'),
+  secretFnCallback: (_req: any, _token: any, cb: any) => { cb(null, 'supersecret') },

```

Comment:
Shouldn't the function arguments should infer the correct types here?
Providing `:any` seems to be breaking changes.

We may type the `const` as `Record<string, FastifyJWTOptions['secret']`

Reply:
Makes sense

Reply:
We have again issues with determining the right function type